### ISSUE TITLE
`jsx-self-closing-comp`: Force a closing tag for HTML tags that aren't void tags

### ISSUE THREAD

**Issue by reteps** [eslint-stylistic/eslint-stylistic#878]
### Clear and concise description of the problem

I want to disallow this syntax:

```html
<i class="foo" />
```

As this element is not a void tag in HTML.

### Suggested solution

This would be similar to https://html-eslint.org/docs/rules/require-closing-tags. Add an option like `"require"` for HTML.

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [x] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by 9romise** [comment]
Sounds reasonable, a `voidElements` option for `html` will be good, what do you think?

**Comment by reteps** [comment]
Sounds good!
